updater/lib/dependabot/updater/group_update_creation.rb

```diff
@@ -238,6 +241,13 @@ def requirements_to_unlock(checker)
         end
       end
 
+      def git_dependency?(dependency)
+        GitCommitChecker.new(
+          dependency: dependency,
+          credentials: job.credentials
+        ).git_dependency?
```

Comment:
I did not love having to instantiate this here, since _most_ `UpdateChecker`s have an instance of this memoized, but it is private and not all of them have it, so sorting through that became a bit of a mess, and I opted to just instantiate a new one, since instantiating objects is cheap, this is not a hot path.